src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+    private static final String SIXTH_LOGIN = "SIXTH_LOGIN";
+    private static final String SEVENTH_LOGIN = "SEVENTH_LOGIN";
+    private static final int MIN_VALID_AGE = 18;
+    private static final int VALID_AGE = 60;
+    private static final int MAX_INVALID_AGE = 17;
+    private static final int NEGATIVE_AGE = -1;
+    private static final String VALID_PASSWORD = "123456";
+    private static final String INVALID_PASSWORD = "12345";
+
+    @BeforeAll
+    public static void setUp() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User();
+        validUser.setLogin(FIRST_LOGIN);
+        validUser.setAge(MIN_VALID_AGE);
+        validUser.setPassword(VALID_PASSWORD);
+        int expectedSize = Storage.people.size() + 1;
+        service.register(validUser);
+        int actualSize = Storage.people.size();
+        assertEquals(expectedSize, Storage.people.size(), "Test failed!"
+                + " Size of storage after adding valid user should be "
+                + expectedSize + ", but was "
+                + actualSize
+        );
+        assertTrue(Storage.people.contains(validUser),
+                "Test failed! Expected storage people contains "
+                + validUser + " = true, but was false");
+    }
+
+    @Test
+    public void register_loginIsNull_notOk() {
+        User nullLoginUser = new User();
+        nullLoginUser.setLogin(null);
+        nullLoginUser.setAge(MIN_VALID_AGE);
+        nullLoginUser.setPassword(VALID_PASSWORD);
+        assertThrows(UserValidationException.class, () -> service.register(nullLoginUser));
+    }
+
+    @Test
+    public void register_passwordIsNull_notOk() {
+        User nullPassUser = new User();
+        nullPassUser.setLogin(SECOND_LOGIN);
+        nullPassUser.setAge(MIN_VALID_AGE);
+        nullPassUser.setPassword(null);
+        assertThrows(UserValidationException.class, () -> service.register(nullPassUser));
+    }
+
+    @Test
+    public void register_ageLessThan18_notOk() {
+        User invAgeUser = new User();
+        invAgeUser.setLogin(FOURTH_LOGIN);
+        invAgeUser.setAge(MAX_INVALID_AGE);
+        invAgeUser.setPassword(VALID_PASSWORD);
+        assertThrows(UserValidationException.class, () -> service.register(invAgeUser));
+    }
+
+    @Test
+    public void register_ageIsGreaterThan18_ok() {
+        User validUser = new User();
+        validUser.setLogin(SIXTH_LOGIN);
+        validUser.setAge(VALID_AGE);
+        validUser.setPassword(VALID_PASSWORD);
+        int expectedSize = Storage.people.size() + 1;
+        service.register(validUser);
+        int actualSize = Storage.people.size();
+        assertEquals(expectedSize, Storage.people.size(), "Test failed!"
+                + " Size of storage after adding valid user should be "
+                + expectedSize + ", but was "
+                + actualSize
+        );
```

Comment:
for test with size you need to create separate test. And check other test with it